getSetType<-function() {
	id=0:20
	symbol=c("c0", "c1","c2", "c2_cgp","c2_cp","c2_biocarta","c2_kegg",
			"c2_reactome","c3","c3_mir","c3_tft","c4","c4_cgn","c4_cm",
			"c5","c5_bp","c5_cc","c5_mf","c6","c7", "hallmark")
	
	name=c("C0: all gene sets",
			"C1: positional gene sets",
			"C2: curated gene sets",
			"C2_CGP: chemical and genetic perturbations",
			"C2_CP: Canonical pathways",
			"C2_CP:BIOCARTA: BioCarta gene sets",
			"C2_CP:KEGG: KEGG gene sets",
			"C2_CP:REACTOME: Reactome gene sets",
			"C3: motif gene sets",
			"C3_MIR: microRNA targets",
			"C3_TFT: transcription factor targets",
			"C4: computational gene sets",
			"C4_CGN: cancer gene neighborhoods",
			"C4_CM: cancer modules",
			"C5: GO gene sets",
			"C5_BP: GO biological process",
			"C5_CC: GO cellular component",
			"C5_MF: GO molecular function",
			"C6: oncogenic signatures",
			"C7: immunologic signatures",
			"hallmark: hallmark gene sets")
	
	description=c("All gene sets",
			"Gene sets corresponding to each human chromosome and each cytogenetic band that has at least one gene. (Cytogenetic locations were parsed from HUGO, October 2006, and Unigene, build 197. When there were conflicts, the Unigene entry was used.) These gene sets are helpful in identifying effects related to chromosomal deletions or amplifications, dosage compensation, epigenetic silencing, and other regional effects.",
			"Gene sets collected from various sources such as online pathway databases, publications in PubMed, and knowledge of domain experts. The gene set page for each gene set lists its source.",
			"Gene sets represent expression signatures of genetic and chemical perturbations. A number of these gene sets come in pairs: an xxx_UP (xxx_DN) gene set representing genes induced (repressed) by the perturbation. The gene set page for each gene set lists the PubMed citation on which it is based.",
			"Gene sets from the pathway databases. Usually, these gene sets are canonical representations of a biological process compiled by domain experts.",
			"Gene sets derived from the BioCarta pathway database, http://www.biocarta.com/genes/index.asp",
			"Gene sets derived from the KEGG pathway database, http://www.genome.jp/kegg/pathway.html",
			"Gene sets derived from the Reactome pathway database, http://www.reactome.org/",
			"Gene sets that contain genes that share a cis-regulatory motif that is conserved across the human, mouse, rat, and dog genomes. The motifs are catalogued (PMID: 15735639) and represent known or likely regulatory elements in promoters and 3'-UTRs. These gene sets make it possible to link changes in a microarray experiment to a conserved, putative cis-regulatory element.",
			"Gene sets that contain genes that share a 3'-UTR microRNA binding motif.",
			"Gene sets that contain genes that share a transcription factor binding site defined in the TRANSFAC (version 7.4, http://www.gene-regulation.com/) database. Each of these gene sets is annotated by a TRANSFAC record.",
			"Computational gene sets defined by mining large collections of cancer-oriented microarray data",
			"Gene sets defined by expression neighborhoods centered on 380 cancer-associated genes (PMID: 14593198). This collection is identical to that previously reported in (PMID: 16199517).",
			"Gene sets defined by Segal et al. 2004 (PMID: 15448693). Briefly, the authors compiled gene sets ('modules') from a variety of resources such as KEGG, GO, and others. By mining a large compendium of cancer-related microarray data, they identified 456 such modules as significantly changed in a variety of cancer conditions.",
			"Gene sets are named by GO term and contain genes annotated by that term. GSEA users: Gene set enrichment analysis identifies gene sets consisting of co-regulated genes; GO gene sets are based on ontologies and do not necessarily comprise co-regulated genes.",
			"Gene sets derived from the Biological Process Ontology (http://www.geneontology.org/GO.process.guidelines.shtml).",
			"Gene sets derived from the Cellular Component Ontology (http://www.geneontology.org/GO.component.guidelines.shtml).",
			"Gene sets derived from the Molecular Function Ontology (http://www.geneontology.org/GO.function.guidelines.shtml).",
			"Gene sets represent signatures of cellular pathways which are often dis-regulated in cancer. The majority of signatures were generated directly from microarray data from NCBI GEO or from internal unpublished profiling experiments which involved perturbation of known cancer genes. In addition, a small number of oncogenic signatures were curated from scientific publications.",
			"Gene sets that represent cell states and perturbations within the immune system. The signatures were generated by manual curation of published studies in human and mouse immunology. For each study, pairwise comparisons of relevant classes were made and genes ranked by mutual information. Gene sets correspond to top or bottom genes (FDR < 0.25 or maximum of 200 genes) for each comparison. This resource is generated as part of the Human Immunology Project Consortium (HIPC; http://www.immuneprofiling.org/).",
			"Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying overlaps between gene sets in other MSigDB collections and retaining genes that display coordinate expression.This collection is an initial release of 50 hallmarks which condense information from over 4,000 original overlapping gene sets from v4.0 MSigDB collections C1 through C6.")
	
	return(data.frame(id=id, symbol=symbol, name=name, description=description, stringsAsFactors = FALSE))	
}
